docker/single-process/README.md

```diff
@@ -0,0 +1,94 @@
+Docker image for Huginn using the production environment and separate container for every process
+=================================================
+
+This image runs a linkable [Huginn](https://github.com/cantino/huginn) instance.
+
+It was inspired by the [official docker container for huginn](https://registry.hub.docker.com/u/cantino/huginn)
+
+The scripts/init script generates a .env file containing the variables as passed as per normal Huginn documentation.
+The same environment variables that would be used for Heroku PaaS deployment are used by this script.
+
+The scripts/init script is aware of mysql and postgres linked containers through the environment variables:
+
+    MYSQL_PORT_3306_TCP_ADDR
+    MYSQL_PORT_3306_TCP_PORT
+
+and
+
+    POSTGRESQL_PORT_5432_TCP_ADDR
+    POSTGRESQL_PORT_5432_TCP_PORT
+
+Its recommended to use an image that allows you to create a database via environmental variables at docker run, like `postgresql` or `mysql`, so the db is populated when this script runs.
+
+Additionally, the database variables may be overridden from the above as per the standard Huginn documentation:
+
+    DATABASE_ADAPTER #(must be either 'postgresql' or 'mysql2')
+    DATABASE_HOST
+    DATABASE_PORT
+
+This script will run database migrations (rake db:migrate) which should be idempotent.
+
+It will also seed the database (rake db:seed) unless this is defined:
+
+    DO_NOT_SEED
+
+This same seeding initially defines the "admin" user with a default password of "password" as per the standard Huginn documentation.
+
+If you do not wish to have the default 6 agents, you will want to set the above environment variable after your initially deploy, otherwise they will be added automatically the next time a container pointing at the database is spun up.
+
+The CMD launches Huginn via the scripts/init script. This may become the ENTRYPOINT later.  It does take under a minute for Huginn to come up.  Use environmental variables that match your DB's creds to ensure it works.
+
+## Usage
+
+Simple startup using docker compose (you need to daemonize with `-d` to persist the data):
+
+    cd docker/single-process
+    docker-compose up
```

Comment:
It might be that the rake processes are not closing their connection to mysql before they terminate, or the mysql logging is just too verboe. I did not notice any issues caused by it though.